File: apps/api/app/routers/billing.py
from fastapi import APIRouter

from app.deps import ManagerContextDep
from app.models.billing import CreditGrant, UsageSummary
from app.services import billing_service
from app.utils.responses import DataResponse, ok

router = APIRouter(tags=["billing"])


@router.get(
    "/workspaces/{workspace_id}/billing/usage",
    response_model=DataResponse[UsageSummary],
)
async def get_billing_usage(ctx: ManagerContextDep) -> DataResponse[UsageSummary]:
    return ok(billing_service.get_usage_summary(ctx.workspace_id))


@router.get(
    "/workspaces/{workspace_id}/billing/grants",
    response_model=DataResponse[list[CreditGrant]],
)
async def list_billing_grants(ctx: ManagerContextDep) -> DataResponse[list[CreditGrant]]:
    return ok(billing_service.list_grants(ctx.workspace_id))

File: apps/web/lib/api/billing.ts
import 'server-only';
import { apiCall } from './client';
import type { CreditGrant, UsageSummary } from '@/lib/types';

export function getBillingUsage(workspaceId: string) {
  return apiCall<UsageSummary>(`/v1/workspaces/${workspaceId}/billing/usage`, {
    next: { revalidate: 60 },
  });
}

export function listBillingGrants(workspaceId: string) {
  return apiCall<CreditGrant[]>(`/v1/workspaces/${workspaceId}/billing/grants`, {
    next: { revalidate: 60 },
  });
}
